src/nedap/speced/def/impl/analysis.cljc

```diff
@@ -109,6 +109,78 @@
                     args-with-proper-tag-hints (ensure-proper-type-hints clj? args)]
                 (apply list args-with-proper-tag-hints prepost body))))))
 
+(defn add-checking-form [tails ret-spec clj?] ;; deserves a test now
```

Comment:
Cover with tests

Particularly, for :pre / :post preservation (while we're there: fix https://github.com/nedap/speced.def/issues/90)